keras_hub/src/models/d_fine/d_fine_layers.py

```diff
@@ -0,0 +1,1796 @@
+        max_num_bins: int, The maximum number of bins for the predictions.
+        lqe_hidden_dim: int, The hidden dimension for the MLP.
+        lqe_layers: int, The number of layers in the MLP.
+        **kwargs: Additional keyword arguments passed to the parent class.
+    """
+
+    def __init__(
+        self,
+        top_prob_values,
+        max_num_bins,
+        lqe_hidden_dim,
+        lqe_layers,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.top_prob_values = top_prob_values
+        self.max_num_bins = max_num_bins
+        self.reg_conf = DFineMLP(
+            input_dim=4 * (self.top_prob_values + 1),
+            hidden_dim=lqe_hidden_dim,
+            output_dim=1,
+            num_layers=lqe_layers,
+            dtype=self.dtype_policy,
+            last_layer_initializer="zeros",
+            name="reg_conf",
+        )
+
+    def build(self, input_shape):
+        reg_conf_input_shape = (
+            input_shape[0][0],
+            input_shape[0][1],
+            4 * (self.top_prob_values + 1),
+        )
+        self.reg_conf.build(reg_conf_input_shape)
+        super().build(input_shape)
+
+    def call(self, scores, pred_corners, training=None):
+        original_shape = keras.ops.shape(pred_corners)
+        batch_size = original_shape[0]
+        length = original_shape[1]
+        reshaped_pred_corners = keras.ops.reshape(
+            pred_corners, (batch_size, length, 4, self.max_num_bins + 1)
+        )
+        prob = keras.ops.softmax(reshaped_pred_corners, axis=-1)
+        prob_topk, _ = keras.ops.top_k(
+            prob, k=self.top_prob_values, sorted=True
+        )
+        stat = keras.ops.concatenate(
+            [prob_topk, keras.ops.mean(prob_topk, axis=-1, keepdims=True)],
+            axis=-1,
+        )
+        reshaped_stat = keras.ops.reshape(stat, (batch_size, length, -1))
+        quality_score = self.reg_conf(reshaped_stat, training=training)
+        return scores + quality_score
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "top_prob_values": self.top_prob_values,
+                "max_num_bins": self.max_num_bins,
+                "lqe_hidden_dim": self.reg_conf.hidden_dim,
+                "lqe_layers": self.reg_conf.num_layers,
+            }
+        )
+        return config
+
+
+@keras.saving.register_keras_serializable(package="keras_hub")
+class DFineConvNormLayer(keras.layers.Layer):
+    """Convolutional layer with normalization and optional activation.
+
+    This is a fundamental building block used in the CNN parts of D-FINE. It
+    combines a `Conv2D` layer with `BatchNormalization` and an optional
+    activation. It is used extensively in layers like `DFineRepVggBlock`,
+    `DFineCSPRepLayer`, and within the `DFineHybridEncoder`.
+
+    Args:
+        in_channels: int, The number of input channels.
+        out_channels: int, The number of output channels.
```

Comment:
Standard convention we use in Keras for output channels is `filters`, you can keep it as `filters`.